homework_1/zad3/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"strings"
+)
+
+type pokemonInformation struct {
+	Name string
+	LocationAreaEncounters string `json:"location_area_encounters"`
+}
+
+type pokemonLocation struct {
+	LocationArea locationArea `json:"location_area"`
+}
+
+type locationArea struct {
+	Name string
+}
+
+const pokemonURL= "https://pokeapi.co/api/v2/pokemon/"
+
+func main() {
+
+	fmt.Println("Pokemon name or number:")
+
+	var pokemon string
+	_, err := fmt.Scanln(&pokemon)
+	if err != nil {
+		log.Fatal(
+			errors.New("Error reading user input."),
+		)
+	}
+
+	httpClient := pester.New()
+	httpResponse, err := httpClient.Get(pokemonURL + pokemon)
```

Comment:
you can extract the http fetch to a separate function. it makes the code cleaner and less verbose

Reply:
additionally, please use proper and safe joining function https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join/34668130